### PR TITLE
Improve the cfg options for octavia ingress controller

### PR DESCRIPTION
**Pull request by lingxiankong**
**What this PR does / why we need it**:
Add/change some options for octavia ingress controller configuration, make most of the options consistent with cloud-config, so it's easy to config octavia-ingress-controller together with openstack-cloud-controller-manager.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

File: pkg/ingress/config/config.go [hunk 1]
```diff
@@ -39,31 +40,50 @@ type kubeConfig struct {
...
 		Password:         cfg.OpenStack.Password,
 		TenantID:         cfg.OpenStack.ProjectID,
-		DomainName:       "default",
+		DomainID:         cfg.OpenStack.DomainID,
+		DomainName:       cfg.OpenStack.DomainName,
+		AllowReauth:      true,
+	}
+}
+
+func (cfg Config) ToV3AuthOptions() tokens.AuthOptions {
```
Comment (ramineni) [inline pkg/ingress/config/config.go]: Openstack now supports only v3 right, so maintaining 2 configs is necessary? ToAuthOptions & ToV3AuthOptions? could you let me know if there is any guideline that cloud-provider-openstack maintains compatibility upto this release of openstack?

Comment (ramineni) [reply pkg/ingress/config/config.go]: @lingxiankong yes, its been removed in queens release. https://docs.openstack.org/releasenotes/keystone/queens.html#other-notes

Comment (lingxiankong) [reply pkg/ingress/config/config.go]: Thanks for the link, so yeah, we don't need to maintain that. But I'd rather do that in a separate PR, thoughts? BTW, just curious, are you using octavia-ingress-controller?

**Pull request merged**